src/Simplex/Chat/Store/Messages.hs

```diff
@@ -724,6 +734,50 @@ getGroupChatPreview_ db user (GroupChatPD _ groupId lastItemId_ stats) = do
     Nothing -> pure []
   pure $ AChat SCTGroup (Chat (GroupChat groupInfo) lastItem stats)
 
+findLocalChatPreview_ :: DB.Connection -> User -> PaginationByTime -> ChatListQuery -> IO [AChatPreviewData]
+findLocalChatPreview_ db User {userId} pagination clq = pure []
+
+getLocalChatPreview_ :: DB.Connection -> User -> ChatPreviewData 'CTLocal -> ExceptT StoreError IO AChat
+getLocalChatPreview_ = error "TODO: getLocalChatPreview_"
+
+-- this function can be changed so it never fails, not only avoid failure on invalid json
+toLocalChatItem :: UTCTime -> ChatItemRow -> Either StoreError (CChatItem 'CTLocal)
```

Comment:
what are the differences between this and toDirectChatItem, except this not having QuoteRow? Could this reuse most of the functionality of the latter? Or are "direct" and "note" chat items also somehow conceptually different?

Reply:
for example, direct chat items already include internal chat items, which can't be quoted. So possibly there could be a toChatItemWtQuote (needs a better name), and toDirectChatItem could reuse it

Reply:
`maybeCIFile`, `ciMeta` and `ciTimed` can be reused even for groups. Not sure if that would be worth collecting all the closure variables to arguments.